icu4c/source/test/intltest/strtest.cpp

```diff
@@ -653,6 +701,20 @@ StringTest::TestStringByteSink() {
     }
 }
 
+void
+StringTest::TestStringByteSinkAppendU8() {
+    // ICU-20984 "mitigate some C++20 char8_t breakages"
+    // For the following APIs there are overloads for both
+    // const char * and const char8_t *.
+    // A u8"string literal" has one type or the other
+    // depending on C++ version and compiler settings.
+    std::string result("abc");
+    StringByteSink<std::string> sink(&result);
+    sink.AppendU8("def", 3);
+    sink.AppendU8(u8"ghijkl", 4);
+    assertEquals("abcdefghij", "abcdef\x67\x68\x69\x6a", result.c_str());
```

Comment:
`\x67` is a funny way to spell `g`; isn't this merely testing the compiler, rather than testing this library?
Alternatively, should line 714 `sink.AppendU8(u8"\u00e9")` and check that the bytes actually appended, even in C++11, are `\xc3\xa9`? (I think that would still be testing the compiler rather than this library, but at least it seems more likely to expose some weird dependency on the current locale or something.)

Reply:
The test has to work on a machine where string literals are not ASCII-based.

`AppendU8("def", 3)` appends system-charset bytes while `AppendU8(u8"ghijkl", 4)` appends UTF-8 bytes. I have to test the bytes that I put in. The sink does not convert.

AFAIK there is not currently a compiler for EBCDIC-based z/OS and whatever "i" is called now, but we don't want to close the door on those yet.